virl2_client/models/authentication.py

```diff
@@ -33,67 +33,98 @@
     from ..virl2_client import ClientLibrary
 
 
+_AUTH_URL = "authenticate"
+
+
 class TokenAuth(httpx.Auth):
     """
-    Initially inspired by:
+    Token-based authentication for an httpx session.
+
+    Inspired by:
     https://requests.readthedocs.io/en/v2.9.1/user/authentication/?highlight=AuthBase#new-forms-of-authentication
-    and modified for httpx based on:
+    Modified for httpx based on:
     https://www.python-httpx.org/advanced/#customizing-authentication
     """
 
     requires_response_body = True
 
     def __init__(self, client_library: ClientLibrary):
+        """
+        Initialize the TokenAuth object with a client library instance.
+
+        :param client_library: A client library instance.
+        """
         self.client_library = client_library
-        self._token: Optional[str] = None
+        self._token: str | None = None
 
     @property
-    def token(self) -> Optional[str]:
+    def token(self) -> str | None:
+        """
+        Return the authentication token. If the token has not been set, it is obtained
+        from the server by calling the API's 'authenticate' endpoint.
+        """
         if self._token is not None:
             return self._token
 
-        auth_url = "authenticate"
-        base_url = self.client_library.session.base_url
+        base_url = self.client_library._session.base_url
```

Comment:
_under access will be also reported by linters, mind you. You could reduce them by having a _session property on this object as well

Reply:
Yeah, this is kind of a problem.
The underscore should definitely be used in a situation like this. The end user should never need to access the session, so it should be hidden from them - and this is the tool we have for that. It signalizes that it is an internal thing, that should have no use to the end user and may be changed at any moment. I don't, however, see any reason for the underscore to restrict us in any way. If anything about it changes, it is me making the change. It also very much has a use internally. So, what benefit could we possibly get from restricting its use between classes in the same module?
IMO, the linters are wrong in this case, and I don't know if there is any way to shut them up.

Reply:
That is not pedantic, you are right, there was no reason for that to be public. Fixed, thanks.
We have many, many cases of using private functions and accessing private attributes of other classes/instances withing the same library, such as `self._client_library._remove_lab_local(event.lab)`, where there is no other solution.